Tests/SwiftParserTest/translated/InitDeinitTests.swift

```diff
@@ -68,21 +68,21 @@ final class InitDeinitTests: XCTestCase {
     AssertParse(
       """
       struct FooStructDeinitializerA {
-        deinit 
-        deinit 1️⃣x 
-        deinit 2️⃣x() 
+        deinit
+        deinit1️⃣ 2️⃣x
+        deinit 3️⃣x()
       }
       """,
       diagnostics: [
         // TODO: Old parser expected error on line 2: expected '{' for deinitializer
         // TODO: Old parser expected error on line 3: deinitializers cannot have a name, Fix-It replacements: 10 - 12 = ''
         // TODO: Old parser expected error on line 3: expected '{' for deinitializer
-        DiagnosticSpec(locationMarker: "1️⃣", message: "unexpected text 'x' before deinitializer"),
+        DiagnosticSpec(locationMarker: "1️⃣", message: "consecutive declarations on a line must be separated by ';'"),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "unexpected text 'x' before deinitializer"),
```

Comment:
Are we missing marking `x` as handled by (1)?

Reply:
No, parsing of the `deinit` with the (1) marker ends before `x` and thus `x` is being considered unexpected text before the next `deinit`.

Reply:
(1) is what I'm finding confusing here. If `x` is being treated as unexpected text then we shouldn't have two decls on the same line, so why do we have that diagnostic?